src/sinks/socket.rs

```diff
@@ -40,9 +40,14 @@ pub enum Mode {
     /// Send over UDP.
     Udp(UdpMode),
 
-    /// Send over a Unix domain socket (UDS).
+    /// Send over a Unix domain socket (UDS), in stream mode.
     #[cfg(unix)]
     Unix(UnixMode),
+
+    /// Send over a Unix domain socket (UDS), in datagram mode. Unavailable on macOS.
+    #[cfg(unix)]
+    #[cfg_attr(target_os = "macos", serde(skip))]
+    UnixDatagram(UnixMode),
```

Comment:
I probably missed some discussion, but why is this unavailable on `macos`? I note that we do support unix datagram sockets for the `socket` source.

Reply:
Ah I see. Could we leave the context about why it is unsupported in the documentation comment? I could see asking myself the same question 6 months from now if I stumble upon this again.